Feature - create Azure Table Storage script (#54)
arcus-azure/arcus.scripting: docs/preview/features/powershell/azure-table-storage.md

---
title: "Scripts related to interacting with Azure Table Storage"
layout: default
---

# Azure Table Storage

## Installation

To have access to the following features, you have to import the module:

```powershell
PS> Import-Module -Name Arcus.Scripting.TableStorage
```

## Create new table in Azure Table Storage on Storage Account

> (Re)Create a Azure Table Storage based to an Storage Account.

Reviewer: ```suggestion
(Re)Create a Table Storage within an Azure Storage Account.
```

| Parameter            | Mandatory | Description                                                                                                     |
| -------------------- | --------- | --------------------------------------------------------------------------------------------------------------- |

> | `ResourceGroupName`  | yes       | The resource group where the Table Storage is located                                                           |

Reviewer: ```suggestion
| `ResourceGroupName`  | yes       | The resource group where the Storage Account is located                                                           |
```

| `StorageAccountName` | yes       | The name of the Storage Account to add the table to                                                             |
| `TableName`          | yes       | The name of the table to add on the Storage Account                                                             |
| `DeleteAndCreate`    | no        | The optional flag to indicate whether or not a possible already existing table should be deleted and re-created |

**Example**

With non-existing table:

```powershell

> PS> Create-AzTableStorageAccountTable -ResourceGroupName "stock" -StorageAccountName "admin" -TableName "products"

Reviewer: Perhaps we can rename this to: `Create-AzTableStorage`.
Or do you think that would be to trimmed down?

Author: Is good for me. Better even. But I wasn't sure of the 'StorageAccount' that was present in the 'old' script and if it had a meaning it was placed in the name. So I left it there.

# Creating table 'products' in the storage account 'admin'..."
```

With existing table and re-create:

```powershell

> PS> Create-AzTableStorageAccountTable -ResourceGroupname "stock" -StorageAccountName "admin" -TableName "products"

Reviewer: Seems like the DeleteAnCreate-flag is missing here.
Is this just missing from the documentation, or is the default behavior of the script to delete and recreate an existing table storage?

Author: No, it's missing from the docs. The default behavior is not to delete+create it.
